data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaCustomProducer.java

```diff
@@ -102,8 +103,8 @@ public void produceRawData(final byte[] bytes, final String key) {
             send(topicName, key, bytes).get();
             topicMetrics.update(producer);
         } catch (Exception e) {
-            // TODO: Metrics
-            LOG.error("Error occurred while publishing {}", e.getMessage());
+            topicMetrics.getNumberOfRawDataSendErrors().increment();
```

Comment:
nit: RawDataSend does not provide good indication on what this counter is about.